src/main/java/org/jenkinsci/test/acceptance/recorder/TestRecorderRule.java

```diff
@@ -83,8 +81,10 @@ private void startRecording(Description des) {
                     (gc, gc.getBounds(), getFileFormat(mimeType),
                             outputFormatForScreenCapture, null, null, diagnostics);
             this.screenRecorder.start();
+        } catch (HeadlessException e) {
+            logger.warn("Test recorder does not work with Headless mode");
+            this.headless = true;
```

Comment:
I thought the same but since what it is really happening is that it is headless I preferred to mark as is. Null is like an error situation, headless is not an error simply it is a valid scenario, that I think that if we run with debug images and with some hacks we would be able to record as well.